src/game_battler.cpp

```diff
@@ -315,6 +316,25 @@ bool Game_Battler::UseSkill(int skill_id) {
 			return false;
 		}
 
+		// Calculate effect:
+		int effect;
+		float mul = GetAttributeMultiplier(skill->attribute_effects);
+		if (mul < 0.5f) {
+			// Determined via testing, the heal is always at least 50%
+			mul = 0.5f;
+		}
+
+		effect = source == nullptr ? skill->power : skill->power +
+			source->GetAtk() * skill->physical_rate / 20 +
+			source->GetSpi() * skill->magical_rate / 40;
+		effect *= mul;
+
+		effect += (effect * Utils::GetRandomNumber(-skill->variance * 10, skill->variance * 10) / 100);
+
+		if (effect < 0)
+			effect = 0;
+
```

Comment:
I really don't like to see this logic duplicated here and in the battle code.

Reply:
We should refactor this skill and item effect logic in some common place. I will do this only after battle PRs are merged.